src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -963,76 +993,136 @@ private Jedis getActiveSentinel() {
 		throw new InvalidDataAccessResourceUsageException("No Sentinel found");
 	}
 
+	private boolean isCreatedOrStopped(@Nullable State state) {
+		return State.CREATED.equals(state) || State.STOPPED.equals(state);
+	}
+
+	private boolean isStarted(@Nullable State state) {
+		return State.STARTED.equals(state);
+	}
+
+	private Jedis createJedis(String hostname, int port, JedisClientConfig config) {
+		return createJedis(newHostAndPort(hostname, port), config);
+	}
+
+	private Jedis createJedis(RedisNode node, JedisClientConfig config) {
+		return createJedis(newHostAndPort(node), config);
+	}
+
+	private Jedis createJedis(HostAndPort hostPort, JedisClientConfig config) {
+		return new Jedis(hostPort, config);
+	}
+
+	private Pool<Jedis> createPool() {
+		return isRedisSentinelAware() ? createRedisSentinelPool(getSentinelConfiguration()) : createRedisPool();
+	}
+
 	private static Set<HostAndPort> convertToJedisSentinelSet(Collection<RedisNode> nodes) {
 
 		if (CollectionUtils.isEmpty(nodes)) {
 			return Collections.emptySet();
 		}
 
 		Set<HostAndPort> convertedNodes = new LinkedHashSet<>(nodes.size());
+
 		for (RedisNode node : nodes) {
 			if (node != null) {
-				convertedNodes.add(new HostAndPort(node.getHost(), node.getPort()));
+				convertedNodes.add(newHostAndPort(node));
 			}
 		}
+
 		return convertedNodes;
 	}
 
-	private int getReadTimeout() {
-		return Math.toIntExact(clientConfiguration.getReadTimeout().toMillis());
+	private static HostAndPort newHostAndPort(String hostname, int port) {
```

Comment:
Why do we have to introduce an abstraction over `new`? This causes noise and makes the code harder to read.